File: prims.c
#include<stdio.h>
#include<stdbool.h>

#define MAX 100
#define INF 9999

void prims(int G[MAX][MAX], int vertices){
    bool selected[MAX];

    // Set selected vertices to false initially
    for (int i = 0; i < MAX; i++) {
        selected[i] = false;
    }

    selected[0] = true;
    int edgeCount = 0;
    int min_cost = 0;

    while (edgeCount < vertices - 1) {
        int min = INF;
        int y = 0;

        for (int i = 0; i < vertices; i++) {
            if (selected[i]) {
                for (int j = 0; j < vertices; j++) {
                    if (!selected[j] && G[i][j] < min) {
                        min = G[i][j];
                        y = j;
                    }
                }
            }
        }
        min_cost += min;
        selected[y] = true;
        edgeCount++;
    }

    printf("%d\n", min_cost);
}

int main() {
    int vertices, edges, u, v, w;
    int G[MAX][MAX];
    
    
    scanf("%d", &vertices);
    scanf("%d", &edges);

    // Initialize adjacency matrix with INF
    for (int i = 0; i < MAX; i++) {
        for (int j = 0; j < MAX; j++) {
            G[i][j] = INF;
        }
    }
    
    // Input weights into adjacency matrix
    for (int i = 0; i < edges; i++) {
        scanf("%d %d %d", &u, &v, &w);
        G[u - 1][v - 1] = w;
    }

    prims(G, vertices);
    return 0;
}

File: prims.py
def prims(graph, vertices):
    selected = [False] * vertices
    selected[0] = True
    edge_count = min_cost = 0
    
    while edge_count < vertices-1:
        min = INF
        s = 0
        for i in range(vertices):
            if selected[i]==True:
                for j in range(vertices):
                    if selected[j]==False and graph[i][j]<min:
                        min = graph[i][j]
                        s = j
        min_cost += min
        selected[s] = True
        edge_count += 1
    
    print(min_cost)
    

vertices = int(input(""))
edges = int(input(""))
INF = 9999
graph = [([INF] * vertices) for i in range(vertices)]

for i in range(edges):
    u, v, weight = map(int, input("").split())
    graph[u-1][v-1] = weight

prims(graph, vertices)